### ISSUE TITLE
Resonance structure generation for O=X makes charged species

### ISSUE THREAD

**Issue by rwest** [cfgoldsmith/RMG-Py#43]
Generating resonance structures of O=X makes [O+]#X which is charged and crashes:

```
m = rmgpy.molecule.Molecule().fromAdjacencyList("""
OX
1 X u0 p0 c0 {2,D}
2 O u0 p2 c0 {1,D}
""")
m
Out[7]:

In [8]:

m.generate_resonance_structures()
---------------------------------------------------------------------------
ResonanceError                            Traceback (most recent call last)
<ipython-input-8-44d7ddb7fde8> in <module>()
----> 1 m.generate_resonance_structures()

rmgpy/molecule/molecule.so in rmgpy.molecule.molecule.Molecule.generate_resonance_structures (build/pyrex/rmgpy/molecule/molecule.c:39755)()

rmgpy/molecule/molecule.so in rmgpy.molecule.molecule.Molecule.generate_resonance_structures (build/pyrex/rmgpy/molecule/molecule.c:39638)()

rmgpy/molecule/resonance.so in rmgpy.molecule.resonance.generate_resonance_structures (build/pyrex/rmgpy/molecule/resonance.c:6352)()

rmgpy/molecule/resonance.so in rmgpy.molecule.resonance.generate_resonance_structures (build/pyrex/rmgpy/molecule/resonance.c:6111)()

rmgpy/molecule/resonance.so in rmgpy.molecule.resonance._generate_resonance_structures (build/pyrex/rmgpy/molecule/resonance.c:7452)()

rmgpy/molecule/resonance.so in rmgpy.molecule.resonance._generate_resonance_structures (build/pyrex/rmgpy/molecule/resonance.c:7265)()

ResonanceError: Resonance generation gave a net charged molecule:
1 O u0 p1 c+1 {2,T}
2 X u0 p0 c0 {1,T}
Ions are not yet supported in RMG.
```

**Comment by rwest** [comment]
1. We could make `Molecule.getNetCharge()` return 0 for all surface species, so the check passes.
2. We could make `resonance.generate_adj_lone_pair_multiple_bond_resonance_structures(mol)` not do anything for surface species.
3. We could make `Atom.updateCharge()` return -1 for the X in the above case (though not sure how it'd know).
4. We could make the check let charged surface species through (though they may cause havoc later)

Do we want O+#X to be generated as a resonance form, or no?

**Comment by cfgoldsmith** [comment]
My immediate thought is that we do not want O+#X generated as a resonance form.

For that matter, I cannot think of any reason why the binding atom should have a charge for any adsorbate (but I do think we want to preserve the capability for charge on other atoms in the adsorbate, e.g. some adsorbate with NO2 hanging off it, however odd that may be).

It may be the case that future versions of RMG-Cat for zeolytes will want to preserve some charge on the binding atom, but right now, no.

So, of the four options, which is least drastic but still does what we want? Perhaps 2?

**Comment by rwest** [comment]
Thanks. Think it's fixed. I made `resonance.generate_adj_lone_pair_multiple_bond_resonance_structures(mol)` skip anything that ends up with a net charge if it's a surface species.  Other parts of the adsorbate should end up with no *net* charge, so should work as you describe.

**Comment by rwest** [comment]
Fixed in 15e4c493cc44e574b8410a889c9e7983ee26b8b8
